# Profitable App profile for App Store market

I work on this project with the guide from [dataquest](https://dataquest.io). I will try to find a mobile app profile that will be profitable on the App Store market. Imaging that I and my team are working as business analysts for a company that make iOS app, our job is to help developers make data-driven decisions on which kind of app they should build.

In [1]:
from csv import reader

opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios_dataset = list(read_file)
ios_dataset_header = ios_dataset[:1]
ios_dataset = ios_dataset[1:]

opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android_dataset = list(read_file)
android_dataset_header = android_dataset[0]
android_dataset = android_dataset[1:]

In the below code, i will define a function to print data sample for any dataset, also inform number of rows and column of entire dataset.

In [2]:
def explore_data(dataset, start, end, rows_and_col=False):
    for row in dataset[start:end]:
        print(row)
        print('\n') #new empty line
    if rows_and_col:
        print('Number of rows: ',len(dataset))
        print('Number of columns: ',len(dataset[0]))
        
#print first 5 row of ios_dataset
print(ios_dataset_header)
print('\n')
explore_data(ios_dataset, 0, 5, 1)

[['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']]


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows:  7197
Number of columns:  16


In [3]:
#print first 5 row of android_dataset
print(android_dataset_header)
print('\n')
explore_data(android_dataset, 0, 5, 1)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

## Delete wrong data

I found [a discussion](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) of the android dataset about missing value in row 10472. I will print some rows around that row for checking.

In [4]:
print(android_dataset_header)
print('\n')
explore_data(android_dataset,10471,10474)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']




As you can see, row 10472 corresponds to the app Life Made WI-Fi Touchscreen Photo Frame having `'Category' = '1.9'`, and `'Rating' = '19'`. Category should not be number and rating cannot over 5. As a consequence, I'll delete this row.

In [5]:
print(len(android_dataset))
del android_dataset[10472]
print(len(android_dataset))

10841
10840


## Clean duplicated entries

### Part one
When reviewing the discussions of both dataset, I found there is at least [one discusion](https://www.kaggle.com/lava18/google-play-store-apps/discussion/67894) about duplicated entries in Android dataset. I checked for Instagram and saw 4 entries:

for row in android_dataset:
    if 'Instagram' in row:
        print(row)


I found there are 1181 cases where apps apear more than once:

In [6]:
def duplicate_check(dataset, key_column = 0):
    duplicate_apps = []
    unique_apps = []
    for row in dataset:
        if row[key_column] in unique_apps:
            duplicate_apps.append(row[key_column])
        else:
            unique_apps.append(row[key_column])
            
    print('Number of duplicate apps:', len(duplicate_apps))
    print('\n')
    print('Examples of duplicate apps:', duplicate_apps[:15])
    return duplicate_apps,unique_apps

android_dup_apps, android_uniq_apps = duplicate_check(android_dataset,0)
print('\n')
#ios_dup_apps, ios_uniq_apps = duplicate_check(ios_dataset,0)

Number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']




When analyzing data, we must not count the same entry more than one. So we will need to remove those duplicated entries. We can remove them randomly, but with this dataset, i found a more effective way.

Examing the duplicated entries of `Instagram` app that were showed above, I found out that the only difference between them is the `review` number. This difference means the data was colected at difference times. We can use it to build a criterion for keeping rows. I will keep the rows that have the highest review number because the higher the number, the newer the data.

To do this, we have to:

* Create a dictionary where each key is app's name and values is the highest review number of that app.
* Use that dictionary to create new data set, which will have only one entry per app and review numbers is the highest


### Part two:

Let start with building the dictionary

In [7]:
reviews_max = {}
for row in android_dataset:
    name = row[0]
    n_reviews = float(row[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

In a previous code, we already found there are 1811 cases where an app occurs more than once, so the lenght of our dictionary should be equal to the difference between the lenght of our dataset and 1181.

In [8]:
print('Length of dictionary: ',len(reviews_max))
print('Expected length: ',len(android_dataset) - 1181)

Length of dictionary:  9659
Expected length:  9659


Now let use the reviews_max to create new dataset that have only one entry per app (with the highest review number) in the code below:

* First, let's create 2 new empty list `android_clean` and `already_add`
* We will loop the `android` dataset, for every iteration:
    * Determine app's `name` and `review number`
    * Add the current row into `android_clean` and `name` into `already_add`, if:
        * Number of review is the highest review. (also mean ignore row don't have highest review number)
        * This app is not in alread_add. (avoid the case that apps have more than one row with highest review number, without this condition there will be apps with more than one entry(both entries also have highest review number)

In [9]:
android_clean = []
already_added = []

for row in android_dataset:
    name = row[0]
    n_reviews = float(row[3])
    
    if n_reviews == reviews_max[name] and name not in already_added:
        android_clean.append(row)
        already_added.append(name)

Let take a quick review of our new android dataset

In [10]:
explore_data(android_clean, 0 , 4, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Number of rows:  9659
Number of columns:  13


It has 9659 row as we calculated before.

## Removing non-English apps

### Part one

While explore data, i found out some apps name is non-English. Here are some examples:

In [11]:
print(ios_dataset[813][1])
print(ios_dataset[6731][1])

print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
中国語 AQリスニング
لعبة تقدر تربح DZ


Our target customers are English-speaking, so we will have to remove these non-English apps. The solution is removing apps that contain non-English character in their names.English text usually includes letters from the English alphabet, numbers composed of digits from 0 to 9, punctuation marks (., !, ?, ;, etc.), and other symbols (+, *, /, etc.).

All these characters that are specific to English texts are encoded using the ASCII standard. Each ASCII character has a corresponding number between 0 and 127 associated with it, and we can take advantage of that to build a function that checks an app name and tells us whether it contains non-ASCII characters.

I create a function to check whether a `string` contain non-English character or not.

In [12]:
def is_English(text):
    for char in text:
        if ord(char) > 127:
            return False
    return True

print(is_English('Instagram'))
print(is_English('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
False


This function work but there are some exeptions will lead to wrong result. For example: some app use `™` or `😜` in there names. We cannot remove these apps

In [13]:
print(is_English('Docs To Go™ Free Office Suite'))
print(is_English('Instachat 😜'))

print(ord('™'))
print(ord('😜'))          

False
False
8482
128540


### Part two

To minimize the impact of data loss, we'll only remove an app if its name has more than 3 non-ASCII characters.

In [14]:
def is_English(text):
    non_ascii = 0;
    for char in text:
        if ord(char) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True
    
print(is_English('Docs To Go™ Free Office Suite'))
print(is_English('Instachat 😜'))

True
True


The function is still not perfect, and very few non-English apps might get past our filter, but this seems good enough at this point in our analysis — we shouldn't spend too much time on optimization at this point.

Let start to filter out non-English app from both 2 data sets:

In [15]:
ios_eng = []
android_eng = []

for row in ios_dataset:
    if is_English(row[1]):
        ios_eng.append(row)
        
for row in android_clean:
    if is_English(row[0]):
        android_eng.append(row)
        
explore_data(ios_eng, 0, 4, True)
print('\n')
explore_data(android_eng, 0, 4, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows:  6183
Number of columns:  16


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Ske

At this point, we have 6183 iOS apps and 9414 Android apps left.

## Removing non-Free apps

The main goal is to create a profitable free app, so we continue to remove non-Free apps from data sets.

In [16]:
ios_final = []
android_final = []

for row in ios_eng:
    price = row[4]
    if price == '0.0':
        ios_final.append(row)
        
for row in android_eng:
    price = row[7]
    if price == '0':
        android_final.append(row)
        
print(len(ios_final))
print(len(android_final))

3222
8864


We have left 3222 iOS apps and 8864 Android apps. It is enough for analyzing

## Most Common Apps by Genre

### Part One

As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

Build a minimal Android version of the app, and add it to Google Play.
If the app has a good response from users, we then develop it further.
If the app is profitable after six months, we also build an iOS version of the app and add it to the App Store.
Because our end goal is to add the app on both the App Store and Google Play, we need to find app profiles that are successful on both markets. For instance, a profile that might work well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of the most common genres for each market. For this, we'll build a frequency table for the prime_genre column of the App Store data set, and the Genres and Category columns of the Google Play data set.

### Part two

We'll build two functions to help analyzing the data:
* One is to calculate frequence for each genre
* Another is to display the frequences in descending order

In [17]:
def freq_cal(data_set, index):
    total_apps = len(data_set)
    freq_table = {}
    for row in data_set:
        if row[index] in freq_table:
            freq_table[row[index]] += 1
        else:
            freq_table[row[index]] = 1

    for key in freq_table:
        freq_table[key] = freq_table[key] / total_apps * 100
        
    return freq_table

def display_percentage(data_set, index):
    table = freq_cal(data_set,index)
    tupled_list = []
    
    for key in table:
        tupled = (key, table[key])
        tupled_list.append(tupled)
        
    sorted_list = sorted(tupled_list, key = lambda app: app[1], reverse = True)
    for row in sorted_list:
        print (row[0], ':', row[1])

### Part three

We start by examining the frequency table for the `prime_genre` column of the App Store data set.

In [18]:
display_percentage(ios_final, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We can see that among the free English apps, more than a half (58.16%) are games. Entertainment apps are close to 8%, followed by photo and video apps, which are close to 5%. Only 3.66% of the apps are designed for education, followed by social networking apps which amount for 3.29% of the apps in our data set.

The general impression is that App Store (at least the part containing free English apps) is dominated by apps that are designed for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are more rare. However, the fact that fun apps are the most numerous doesn't also imply that they also have the greatest number of users — the demand might not be the same as the offer.

Let's continue by examining the `Genres` and `Category` columns of the Google Play data set (two columns which seem to be related).

In [19]:
display_percentage(android_final, 1) #Category

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

The landscape seems significantly different on Google Play: there are not that many apps designed for fun, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.). However, if we investigate this further, we can see that the family category (which accounts for almost 19% of the apps) means mostly games for kids.

Even so, practical apps seem to have a better representation on Google Play compared to App Store. This picture is also confirmed by the frequency table we see for the `Genres` column:

In [20]:
display_percentage(android_final, -4) #Genres

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The difference between the `Genres` and the `Category` columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the `Category` column moving forward.

Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. Now we'd like to get an idea about the kind of apps that have most users.

## Most Popular apps on App Store

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the `Installs` column, but for the App Store data set this information is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the `rating_count_tot` app.

Below, we calculate the average number of user ratings per app genre on the App Store:

In [21]:
ios_genres = freq_cal(ios_final, -5)

for genre in ios_genres:
    total = 0
    len_genre = 0
    for row in ios_final:
        if row[-5] == genre:
            total += float(row[5])
            len_genre += 1
    average = total / len_genre
    print(genre,':',average)           
    

Navigation : 86090.33333333333
Photo & Video : 28441.54375
News : 21248.023255813954
Utilities : 18684.456790123455
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Business : 7491.117647058823
Sports : 23008.898550724636
Social Networking : 71548.34905660378
Medical : 612.0
Productivity : 21028.410714285714
Music : 57326.530303030304
Entertainment : 14029.830708661417
Shopping : 26919.690476190477
Food & Drink : 33333.92307692308
Travel : 28243.8
Book : 39758.5
Catalogs : 4004.0
Games : 22788.6696905016
Lifestyle : 16485.764705882353
Finance : 31467.944444444445
Education : 7003.983050847458
Reference : 74942.11111111111


On average, navigation apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together:

In [22]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1],':',app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


The same pattern applies to Social Networking, where the most popular are from giant like Facebook, Skype, Pinterest, etc. It also the same with music app, few biggest apps got almost the attention. These genres are influenced by big players.

Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold. We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.

After dig deep into each genre, We found out that Reference, Productivity and especially Games are very promising:
* On Reference genre has high average user ratings, but mostly because Bible and Dictionary app skew up the number, other apps also have high average user rating, but however the dominant apps are book-related apps, difficult to come up with new idea.
* In Productivity genre, there is variety kind of apps, and seem pretty balance between apps.
* Games genre has a lot of really popular apps, but it seem like too saturated. Hard to stand out among huge number of apps.

Sum up, from these statistic, It is very potential to make an app that relate to  these features: fun, helpfull, practical.
For example, we could make a schedule reminder apps that fun to use (kind of funny animated when complete an task, or mini-games to play), maybe a studying app that designed like a game (have level, point when complete a mission...)

In [34]:
for app in ios_final:
    if app[-5] == 'Productivity':
        print(app[1],':',app[5])

Evernote - stay organized : 161065
Gmail - email by Google: secure, fast & organized : 135962
iTranslate - Language Translator & Dictionary : 123215
Yahoo Mail - Keeps You Organized! : 113709
Google Docs : 64259
Google Drive - free online storage : 59255
Dropbox : 49578
Microsoft Word : 47999
Microsoft OneNote : 39638
Microsoft Outlook - email and calendar : 32807
Hotspot Shield Free VPN Proxy & Wi-Fi Privacy : 32499
Documents 6 - File manager, PDF reader and browser : 29110
Google Sheets : 24602
Microsoft Excel : 24430
Inbox by Gmail : 21561
T-Mobile : 19977
Paper by FiftyThree - Sketch, Diagram, Take Notes : 18219
MyScript Calculator - Handwriting calculator : 16555
VPN Proxy Master - Unlimited WiFi security VPN : 13674
Microsoft OneDrive – File & photo cloud storage : 12797
Ever - Capture Your Memories : 12755
Speak & Translate － Voice and Text Translator : 12062
Tayasui Sketches : 11505
Drawing Desk - Draw, Paint, Doodle & Sketch board : 11040
Microsoft PowerPoint : 10939
Email - F

Beside that, we need to avoid some genres like:
* Weather: could be costly to integrate with a highly accurate prediction -api
* Music: very troublesome when involve with copyright law if we don't have huge financial resoures
* Finance: we need to hire an expert in this field.
* Food & Drink: mostly the apps are from popular food chains

Now let's analyze the Google Play market a bit.

## Most Popular Apps by Genre on Google Play

For the Google Play market, we actually have data about the number of installs, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [24]:
display_percentage(android_final, 5) #Install

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


One problem with this data is that is not precise. For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to get an idea which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

To perform computations, however, we'll need to convert each install number to float — this means that we need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error. We'll do this directly in the loop below, where we also compute the average number of installs for each genre (category).

In [27]:
android_categories = freq_cal(android_final,1)

for category in android_categories:
    total = 0
    len_category = 0
    for app in android_final:
        if app[1] == category:
            install = app[5].replace('+','')
            install = float(install.replace(',',''))
            total += install
            len_category += 1
    avg_intall = total / len_category
    print(category,':', avg_intall)

ART_AND_DESIGN : 1986335.0877192982
COMMUNICATION : 38456119.167247385
LIBRARIES_AND_DEMO : 638503.734939759
PRODUCTIVITY : 16787331.344927534
EVENTS : 253542.22222222222
TRAVEL_AND_LOCAL : 13984077.710144928
AUTO_AND_VEHICLES : 647317.8170731707
MAPS_AND_NAVIGATION : 4056941.7741935486
WEATHER : 5074486.197183099
PHOTOGRAPHY : 17840110.40229885
FINANCE : 1387692.475609756
SOCIAL : 23253652.127118643
HOUSE_AND_HOME : 1331540.5616438356
BOOKS_AND_REFERENCE : 8767811.894736841
SHOPPING : 7036877.311557789
EDUCATION : 1833495.145631068
DATING : 854028.8303030303
BUSINESS : 1712290.1474201474
LIFESTYLE : 1437816.2687861272
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_MAGAZINES : 9549178.467741935
PERSONALIZATION : 5201482.6122448975
COMICS : 817657.2727272727
BEAUTY : 513151.88679245283
MEDICAL : 120550.61980830671
TOOLS : 10801391.298666667
FOOD_AND_DRINK : 1924897.7363636363
SPORTS : 3638640.1428571427
GAME : 15588015.603248259
PARENTING : 542603.6206896552
HEALTH_AND_FITNESS : 4188821.98

On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [29]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+' 
                                      or app[5] == '500,000,000+' 
                                      or app[5] == '100,000,000+'):
        print(app[0],':',app[5])
        

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

If we removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times:

In [31]:
comm_cate_under_100m = []
for app in android_final:
    install = app[5].replace('+','')
    install = float(install.replace(',',''))
    if app[1] == 'COMMUNICATION' and install < 100000000:
        comm_cate_under_100m.append(install)

sum(comm_cate_under_100m) / len(comm_cate_under_100m)

3603485.3884615386

We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

The `PRODUCTIVITY` genre again looks fairly popular as well, with an average number of installs of 16,787,331. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:

In [33]:
for app in android_final:
    if app[1] == 'PRODUCTIVITY':
        print(app[0],':',app[5])

Microsoft Word : 500,000,000+
All-In-One Toolbox: Cleaner, Booster, App Manager : 10,000,000+
AVG Cleaner – Speed, Battery & Memory Booster : 10,000,000+
QR Scanner & Barcode Scanner 2018 : 10,000,000+
Chrome Beta : 10,000,000+
Microsoft Outlook : 100,000,000+
Google PDF Viewer : 10,000,000+
My Claro Peru : 5,000,000+
Power Booster - Junk Cleaner & CPU Cooler & Boost : 1,000,000+
Google Assistant : 10,000,000+
Microsoft OneDrive : 100,000,000+
Calculator - unit converter : 50,000,000+
Microsoft OneNote : 100,000,000+
Metro name iD : 10,000,000+
Google Keep : 100,000,000+
Archos File Manager : 5,000,000+
ES File Explorer File Manager : 100,000,000+
ASUS SuperNote : 10,000,000+
HTC File Manager : 10,000,000+
MyMTN : 1,000,000+
Dropbox : 500,000,000+
ASUS Quick Memo : 10,000,000+
HTC Calendar : 10,000,000+
Google Docs : 100,000,000+
ASUS Calling Screen : 10,000,000+
lifebox : 5,000,000+
Yandex.Disk : 5,000,000+
Content Transfer : 5,000,000+
HTC Mail : 10,000,000+
Advanced Task Killer : 50

The PRODUCTIVITY genre includes a variety of apps: utility like files storage, task manager, reminder, calculator. etc. There is also an amount of more popular  skew up the average installer:

In [35]:
for app in android_final:
    if app[1] == 'PRODUCTIVITY' and (app[5] == '1,000,000,000+' 
                                      or app[5] == '500,000,000+' 
                                      or app[5] == '100,000,000+'):
        print(app[0],':',app[5])

Microsoft Word : 500,000,000+
Microsoft Outlook : 100,000,000+
Microsoft OneDrive : 100,000,000+
Microsoft OneNote : 100,000,000+
Google Keep : 100,000,000+
ES File Explorer File Manager : 100,000,000+
Dropbox : 500,000,000+
Google Docs : 100,000,000+
Microsoft PowerPoint : 100,000,000+
Samsung Notes : 100,000,000+
SwiftKey Keyboard : 100,000,000+
Google Drive : 1,000,000,000+
Adobe Acrobat Reader : 100,000,000+
Google Sheets : 100,000,000+
Microsoft Excel : 100,000,000+
WPS Office - Word, Docs, PDF, Note, Slide & Sheet : 100,000,000+
Google Slides : 100,000,000+
ColorNote Notepad Notes : 100,000,000+
Evernote – Organizer, Planner for Notes & Memos : 100,000,000+
Google Calendar : 500,000,000+
Cloud Print : 500,000,000+
CamScanner - Phone PDF Creator : 100,000,000+


However it seem not to overly - focus on a particular apps, it spread widely between apps, let dig a little more on apps between 1,000,000 and 100,000,000 installers:

In [36]:
for app in android_final:
    if app[1] == 'PRODUCTIVITY' and (app[5] == '1,000,000+'
                                        or app[5] == '5,000,000+'
                                        or app[5] == '10,000,000+'
                                        or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

All-In-One Toolbox: Cleaner, Booster, App Manager : 10,000,000+
AVG Cleaner – Speed, Battery & Memory Booster : 10,000,000+
QR Scanner & Barcode Scanner 2018 : 10,000,000+
Chrome Beta : 10,000,000+
Google PDF Viewer : 10,000,000+
My Claro Peru : 5,000,000+
Power Booster - Junk Cleaner & CPU Cooler & Boost : 1,000,000+
Google Assistant : 10,000,000+
Calculator - unit converter : 50,000,000+
Metro name iD : 10,000,000+
Archos File Manager : 5,000,000+
ASUS SuperNote : 10,000,000+
HTC File Manager : 10,000,000+
MyMTN : 1,000,000+
ASUS Quick Memo : 10,000,000+
HTC Calendar : 10,000,000+
ASUS Calling Screen : 10,000,000+
lifebox : 5,000,000+
Yandex.Disk : 5,000,000+
Content Transfer : 5,000,000+
HTC Mail : 10,000,000+
Advanced Task Killer : 50,000,000+
MyVodafone (India) - Online Recharge & Pay Bills : 10,000,000+
Microsoft Translator : 5,000,000+
My Airtel-Online Recharge, Pay Bill, Wallet, UPI : 50,000,000+
Do It Later: Tasks & To-Dos : 50,000,000+
Verizon Cloud : 50,000,000+
myAT&T : 50,

This niche seems haven't yet been dominated by any particular kind of apps.
However the number of softwares for notes, reminder, calender may be overwhelming. We will need to avoid this subject if we don't have any really out standing concept.

# Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that making a productivily apps will be more potentail. We could think about kinds of life-hack, financial managerment, or any kind of utility apps that help user to handle their lift better. We also can add the fun elements into the app for more appealing to user.